File: src/main/scala/io/izzel/taboolib/util/item/inventory/stored/Action.java
package io.izzel.taboolib.util.item.inventory.stored;

import io.izzel.taboolib.util.item.inventory.ClickEvent;
import org.bukkit.inventory.ItemStack;

public abstract class Action {

    public abstract ItemStack getCurrent(ClickEvent e);

    public abstract void setCurrent(ClickEvent e, ItemStack item);

    public abstract int getCurrentSlot(ClickEvent e);
}

File: module/module-ui/src/main/kotlin/taboolib/module/ui/type/Action.kt
package taboolib.module.ui.type

import org.bukkit.inventory.ItemStack
import taboolib.module.ui.ClickEvent

abstract class Action {

    abstract fun getCursor(e: ClickEvent): ItemStack?

    abstract fun setCursor(e: ClickEvent, item: ItemStack?)

    abstract fun getCurrentSlot(e: ClickEvent): Int
}